Explorer/Assets/Scripts/ECS/StreamableLoading/NftShapes/Urns/BasedUrnSource.cs

```diff
@@ -0,0 +1,41 @@
+using CommunicationData.URLHelpers;
+using DCL.Diagnostics;
+using System;
+using System.Linq;
+
+namespace ECS.StreamableLoading.NftShapes.Urns
+{
+    public class BasedUrnSource : IUrnSource
+    {
+        private readonly string baseUrl;
+
+        public BasedUrnSource(string baseUrl = "https://opensea.decentraland.org/api/v2/chain/ethereum/contract/{address}/nfts/{id}")
+        {
+            this.baseUrl = baseUrl;
+        }
+
+        public URLAddress UrlOrEmpty(string urn)
+        {
+            //"https://opensea.decentraland.org/api/v2/chain/ethereum/contract/0x06012c8cf97bead5deae237070f9587f8e7a266d/nfts/1631847"
+            //urn:decentraland:ethereum:erc721:0x06012c8cf97bead5deae237070f9587f8e7a266d:1631847
+
+            try
+            {
+                string[] array = urn.Split(':')!.TakeLast(2).ToArray();
```

Comment:
did you try to optimize with using Span's?

Reply:
good mention, it's a niec option to avoid allocations

Reply:
New implementation is included